include/xgboost/learner.h

```diff
@@ -167,6 +184,8 @@ class Learner : public Model, public Configurable, public rabit::Serializable {
   virtual std::vector<std::string> DumpModel(const FeatureMap& fmap,
                                              bool with_stats,
                                              std::string format) const = 0;
+
+  virtual XGBAPIThreadLocalEntry& GetThreadLocal() const = 0;
```

Comment:
We could make Learner a concrete class. I don't see us subclassing different learners any time soon.

Reply:
Actually I don't like Learner, as linear and tree are so different.